EncountifyAPI/Controllers/LocationsController.cs

```diff
@@ -0,0 +1,222 @@
+﻿using System;
+using System.Globalization;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using EncountifyAPI.Data;
+using EncountifyAPI.Models;
+using System.Data.SqlClient;
+using Microsoft.Extensions.Configuration;
+
+namespace EncountifyAPI.Controllers
+{
+    [Route("API/[controller]")]
+    [ApiController]
+    public class LocationsController : ControllerBase
+    {
+        private readonly string ConnectionString;
+
+        public LocationsController(IConfiguration configuration)
+        {
+            ConnectionString = configuration.GetConnectionString("EncountifyAPIContext");
+        }
+
+        /// <summary>
+        /// Get all locations
+        /// </summary>
+        [HttpGet]
+        public IEnumerable<Location> GetAllLocations()
+        {
+            return ExecuteReader("SELECT * FROM Locations");
+        }
+
+        /// <summary>
+        /// Get location with specified id
+        /// </summary>
+        [HttpGet("Id/{id}")]
+        public IEnumerable<Location> GetLocation(int id)
+        {
+            return ExecuteReader("SELECT * FROM Locations WHERE Id = @id", id: id);
+        }
+
+        /// <summary>
+        /// Get location with specified location name
+        /// </summary>
+        [HttpGet("Name/{locationname}")]
+        public IEnumerable<Location> GetLocation(string name)
+        {
+            return ExecuteReader("SELECT * FROM Locations WHERE CONVERT(VARCHAR, Name) = @name", name: name);
+        }
+
+        /// <summary>
+        /// Add a new location
+        /// </summary>
+        [HttpPost]
+        public IEnumerable<Location> AddLocation(string name, string description = "", float longitude = 0, float latitude = 0, int category = 0, string image = "")
+        {
+
+            ExecuteQuery("INSERT INTO Locations VALUES (@name, @description, @longitude, @latitude, @category, @image)", name: name, description: description, longitude: longitude, latitude: latitude, category: category, image:image);
```

Comment:
I am getting this 
![Screenshot (97)](https://user-images.githubusercontent.com/50084966/142046041-f8c3f039-4819-468d-9582-ca1004e79aa5.png)